designs/2022-lintFiles-lintText-fix-support/README.md

```diff
@@ -0,0 +1,251 @@
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+### eslint.lintText(code, options)
+
+The options of `lintText(code, options)` support the fix parameter and pass it to `cliEngine.executeOnText()`:
+
+```javascript
+async lintText(code, options = {}) {
+    // ...
+
+    const {
+        filePath,
+        warnIgnored = false,
+        fix,
+        ...unknownOptions
+    } = options || {};
+
+    // ...
+
+    const { cliEngine } = privateMembersMap.get(this);
+
+    return processCLIEngineLintReport(
+        cliEngine,
+        cliEngine.executeOnText(code, filePath, warnIgnored, fix)
+    );
+}
+```
+
+`cliEngine.executeOnText` merges the fix passed in `lintText()` and the `options.fix` in `ESLint.constructor()` (I omit irrelevant code):
+
+```javascript
+executeOnText(text, filename, warnIgnored, fixFromLintText) {
+    const {
+        options: {
+            fix: fixFromESLint,
+        }
+    } = internalSlotsMap.get(this);
+
+        // Do lint.
+        results.push(verifyText({
+            // text,
+            // filePath: resolvedFilename,
+            // config,
+            // cwd,
+            fix: fixFromLintText !== undefined ? fixFromLintText : fixFromESLint,
+            // allowInlineConfig,
+            // reportUnusedDisableDirectives,
+            // fileEnumerator,
+            // linter
+        }));
+}
+```
+
+### eslint.lintFiles(patterns)
+
+`eslint.lintFiles(patterns)` adds options parameter and provides fix parameter:
+
+```javascript
+async lintFiles(patterns, options = {}) {
+    // ...
+
+    const {
+        fix,
+        ...unknownOptions
+    } = options || {};
+
+    const unknownOptionKeys = Object.keys(unknownOptions);
+
+    if (unknownOptionKeys.length > 0) {
+        throw new Error(`'options' must not include the unknown option(s): ${unknownOptionKeys.join(", ")}`);
+    }
+
+    const { cliEngine } = privateMembersMap.get(this);
+
+    return processCLIEngineLintReport(
+        cliEngine,
+        cliEngine.executeOnFiles(patterns, fix)
+    );
+}
+```
+
+`cliEngine.executeOnFiles` has basically the same changes as `cliEngine.executeOnText` above:
+
+```javascript
+executeOnFiles(text, fixFromLintFiles) {
+    const {
+        options: {
+            fix: fixFromESLint,
+        }
+    } = internalSlotsMap.get(this);
+
+        // Do lint.
+        results.push(verifyText({
+            // text,
+            // filePath: resolvedFilename,
+            // config,
+            // cwd,
+            fix: fixFromLintFiles !== undefined ? fixFromLintFiles : fixFromESLint,
+            // allowInlineConfig,
+            // reportUnusedDisableDirectives,
+            // fileEnumerator,
+            // linter
+        }));
+}
+```
+
+## Documentation
+
+No.
```

Comment:
Yes :) We need documentation for API changes. Please figure out where the documentation needs to be changed to explain this feature.